Code/Framework/AzCore/Tests/Asset/AssetManagerLoadingTests.cpp

```diff
@@ -3171,6 +3182,99 @@ namespace UnitTest
         EXPECT_TRUE(nestedDependentAsset.IsReady());
     }
 
+    TEST_F(AssetManagerClearAssetReferenceTests, ReloadTest)
+    {
+        // Regression test - there was a bug where rapid reloads could get stuck due to the owning container being invalidated
+        // Note that for this bug to occur, the loaded asset needs to have dependencies
+        // Order of events to reproduce:
+        // 1) Asset is loaded
+        // 2) Reload occurs
+        // 3) Another reload begins
+        // 4) Old asset is reassigned
+        // 4a) Old asset ref count hits 0
+        // 4b) Old asset triggers OnAssetUnused
+        // 4c) Container is released
+        // 5) Reload stalls because container is gone
```

Comment:
are these the steps that got fixed?
if so, should include the new expected order here?

Reply:
Updated it a bit to point out where the fix is